src/ConnectedMode/UI/ProjectSelection/ProjectSelectionWindow.xaml.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Diagnostics.CodeAnalysis;
+using System.Windows;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UI.ProjectSelection;
+
+[ExcludeFromCodeCoverage]
+public partial class ProjectSelectionWindow
+{
+    public ProjectSelectionViewModel ViewModel { get; }
+
+    public ProjectSelectionWindow(ConnectionInfo.Connection connection)
+    {
+        ViewModel = new ProjectSelectionViewModel(connection);
+        
+        InitializeComponent();
+
+        ViewModel.BindingFinished += (_, _) => Close();
```

Comment:
Please make sure to unsubscribe from the ViewModel's event to prepare memory leaks.

You should create a CloseWindow() event handler, register to it in the constructor and then unregister from it when the window is unloaded, for example (Wpf components have an Unloaded event), or any other appropriate event.